	* gio/gdbusmessage.c (g_dbus_message_print): Fix compilation error
	on Solaris.

	* gio/Makefile.in: Disable build in subdirectory tests, which does
	not build out of the box on Solaris

*** gio/Makefile.in.original	Tue Feb 16 17:28:32 2010
--- gio/Makefile.in	Tue Feb 16 17:28:56 2010
*************** ETAGS = etags
*** 258,264 ****
  CTAGS = ctags
  am__tty_colors = \
  red=; grn=; lgn=; blu=; std=
! DIST_SUBDIRS = libasyncns xdgmime inotify fen win32 . fam tests
  DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
  am__relativize = \
    dir0=`pwd`; \
--- 258,264 ----
  CTAGS = ctags
  am__tty_colors = \
  red=; grn=; lgn=; blu=; std=
! DIST_SUBDIRS = libasyncns xdgmime inotify fen win32 . fam 
  DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
  am__relativize = \
    dir0=`pwd`; \
***************
*** 602,608 ****
  TEST_PROGS = 
  NULL = 
  SUBDIRS = $(am__append_1) $(am__append_3) $(am__append_7) \
! 	$(am__append_10) . $(am__append_13) tests
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
--- 602,608 ----
  TEST_PROGS = 
  NULL = 
  SUBDIRS = $(am__append_1) $(am__append_3) $(am__append_7) \
! 	$(am__append_10) . $(am__append_13) 
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@noinst_DATA = gio-2.0.lib
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@install_ms_lib_cmd = $(INSTALL) gio-2.0.lib $(DESTDIR)$(libdir)
  @MS_LIB_AVAILABLE_TRUE@@OS_WIN32_AND_DLL_COMPILATION_TRUE@uninstall_ms_lib_cmd = -rm $(DESTDIR)$(libdir)/gio-2.0.lib
--- gio/gdbusmessage.c.old	2010-11-12 14:52:52.000000000 +0100
+++ gio/gdbusmessage.c	2010-12-08 09:02:42.028479578 +0100
@@ -3081,6 +3081,12 @@ g_dbus_message_print (GDBusMessage *mess
     }
   g_string_append (str, "\n");
 #ifdef G_OS_UNIX
+
+#ifndef major
+#define major(x) 0
+#define minor(x) 0
+#endif
+
   g_string_append_printf (str, "%*sUNIX File Descriptors:\n", indent, "");
   if (message->fd_list != NULL)
     {
